maak een nieuw blogbericht over de Prijsherzieningsindex I 2021

De webpagina voor de data is: https://economie.fgov.be/nl/themas/ondernemingen/specifieke-sectoren/bouw/prijsherzieningsindexen/mercuriale-index-i-2021

de data is hier: https://economie.fgov.be/sites/default/files/Files/Entreprises/prix-construction-Indice-I-2021.xlsx

de methodologie is hier: https://economie.fgov.be/sites/default/files/Files/Entreprises/Hervorming-van-de-Mercuriale-Methodologische-Nota.pdf

de formules over hoe index I toe te passen is hier: https://economie.fgov.be/sites/default/files/Files/Entreprises/Mercuriale-Hoe-de-index-I-2021-toepassen-in-een-prijsherzieningsformule.pdf

maak een blogbericht :
met uitleg wat de index I is en waarvoor die dient
met de grafiek en tabel van de index waarbij je kan filteren op de onderdelen, vereenvoudig de onderdelen ook hun naam
een tool waarmee je prijsherzieningsformule kan toepassen
csv en iframe downloadknoppen waar mogelijk
 zorg voor automatische maandelijkse update van de data via GitHub Actions
geef weer wanneer de data laatst werd bijgewerkt

@claude ik heb de yaml toegevoegd, maar ik krijg deze error: 52s
Run npx --no-install next build
⚠ No build cache found. Please configure build caching for faster rebuilds. Read more: https://nextjs.org/docs/messages/no-cache
Attention: Next.js now collects completely anonymous telemetry regarding usage.
This information is used to shape Next.js' roadmap and prioritize features.
You can learn more, including how to opt-out if you'd not like to participate in this anonymous program, by visiting the following URL:
https://nextjs.org/telemetry

  ▲ Next.js 14.2.13

   Creating an optimized production build ...
Generated 9 documents in .contentlayer
<w> [webpack.cache.PackFileCacheStrategy/webpack.FileSystemInfo] Parsing of /home/runner/work/data-blog/data-blog/embuild-analyses/node_modules/@contentlayer/core/dist/generation/generate-dotpkg.js for build dependencies failed at 'import(filePathJoin(generatedPkgPath, 'generated', 'index.mjs'))'.
<w> Build dependencies behind this expression are ignored and might cause incorrect cache invalidation.
<w> [webpack.cache.PackFileCacheStrategy/webpack.FileSystemInfo] Parsing of /home/runner/work/data-blog/data-blog/embuild-analyses/node_modules/@contentlayer/core/dist/generation/generate-dotpkg.js for build dependencies failed at 'import(filePathJoin(generatedPkgPath, 'generated', 'index.mjs'))'.
<w> Build dependencies behind this expression are ignored and might cause incorrect cache invalidation.
<w> [webpack.cache.PackFileCacheStrategy/webpack.FileSystemInfo] Parsing of /home/runner/work/data-blog/data-blog/embuild-analyses/node_modules/@contentlayer/core/dist/generation/generate-dotpkg.js for build dependencies failed at 'import(filePathJoin(generatedPkgPath, 'generated', 'index.mjs'))'.
<w> Build dependencies behind this expression are ignored and might cause incorrect cache invalidation.
 ✓ Compiled successfully
   Linting and checking validity of types ...
Failed to compile.

./src/components/analyses/faillissementen/FaillissementenDashboard.tsx:279:3
Type error: Property 'sectors' is missing in type '{ data: Sector[]; error: string | null; }' but required in type '{ sectors: Sector[]; error: string | null; }'.

  277 |     return { sectors: lookups.sectors, error: null }
  278 |   }
> 279 |   return safeArrayAccess<Sector>(
      |   ^
  280 |     (lookups as Record<string, unknown>).sectors,
  281 |     'sectorgegevens'
  282 |   )
Error: Process completed with exit code 1.

--- a/embuild-analyses/src/components/analyses/faillissementen/FaillissementenDashboard.tsx
+++ b/embuild-analyses/src/components/analyses/faillissementen/FaillissementenDashboard.tsx
@@ -276,20 +276,22 @@ function useSectorOptions(): { sectors: Sector[]; error: string | null } {
   if (isValidLookups(lookups)) {
     return { sectors: lookups.sectors, error: null }
   }
-  return safeArrayAccess<Sector>(
+  const { data, error } = safeArrayAccess<Sector>(
     (lookups as Record<string, unknown>).sectors,
     'sectorgegevens'
   )
+  return { sectors: data, error }
 }
 
 function useProvinceOptions(): { provinces: Province[]; error: string | null } {
   if (isValidLookups(lookups)) {
     return { provinces: lookups.provinces, error: null }
   }
-  return safeArrayAccess<Province>(
+  const { data, error } = safeArrayAccess<Province>(
     (lookups as Record<string, unknown>).provinces,
     'provinciegegevens'
   )
+  return { provinces: data, error }
 }
 
 // Sector filter dropdown

